pillar/cni.sls

```diff
@@ -1,6 +1,6 @@
 # the flannel backend ('udp', 'vxlan', 'host-gw', etc)
 flannel:
-  image:          'sles12/flannel:0.9.1'
+  image:          '{{ grains["base_image_url"] }}/flannel:0.9.1'
```

Comment:
Isn't a pillar more suitable for this than a grain? (regarding `base_image_url`). A grain is machine-oriented, but I think this setting is more cluster-wide, right?

Reply:
I 100% agree with you that it *should* be a pillar.

The problem is that I couldn't get it to work, as I would have to access the pillar in other pillars, which seems not possible in salt.
I tried setting a variable as well `{% set some_pillar = 'some_value' %}` and use that instead, but that also doesn't work, so I resorted to using a grain as it was the only solution that worked.

If there is a way to use a pillar instead I would love to use it.

Reply:
Yeah, that is a known problem with pillars: you cannot call a pillaar from a pillar. Maybe you could try to call a `_module/*` function... I don't know.